firebase-functions/src/main/java/com/google/firebase/functions/HttpsCallableReference.kt

```diff
@@ -150,8 +168,21 @@ public class HttpsCallableReference {
    * @param units The units for the specified timeout.
    */
   public fun withTimeout(timeout: Long, units: TimeUnit): HttpsCallableReference {
-    val other = HttpsCallableReference(functionsClient, name, options)
+    val other = HttpsCallableReference(functionsClient, name, options, headers)
     other.setTimeout(timeout, units)
     return other
   }
+
+  /**
+   * Adds an HTTP header for calls from this instance of Functions.
+   *
+   * Note that an existing header with the same name will be overwritten.
+   *
+   * @param name Name of HTTP header
+   * @param value Value of HTTP header
+   */
+  public fun addHeader(name: String, value: String): HttpsCallableReference {
```

Comment:
Shall we also add a `addHeaders(headers: Map<String, String>)` function?

Reply:
Sounds good!

Reply:
Done.